benchmark/src/main/scala/fs2/benchmark/StreamBenchmark.scala

```diff
@@ -37,8 +37,8 @@ class StreamBenchmark extends BenchmarkUtils {
     (Stream.chunk(Chunk.seq(0 to 256000)).pure).repeatPull { s =>
       for {
         s2 <- Pull.awaitN(N)(s)
-        _  <- Pull.output(s2.head.head)
-      } yield s2.tail
+        _  <- Pull.output(s2._1.head)
+      } yield s2._2
```

Comment:
I am just wondering if we can implement `filter` on pull and instead of s2 have (_,s) on the left side of for? WDYT?